modules/transactionpool/update.go

```diff
@@ -60,60 +171,69 @@ func (tp *TransactionPool) ProcessConsensusChange(cc modules.ConsensusChange) {
 			}
 		}
 
-		// Add the transactions from this block.
-		var totalSize uint64
-		for _, txn := range block.Transactions {
+		// Find the median transaction fee for this block.
+		type feeSummary struct {
+			fee  types.Currency
+			size int
+		}
+		var fees []feeSummary
+		var totalSize int
+		txnSets := findSets(block.Transactions)
+		for _, set := range txnSets {
+			// Compile the fees for this set.
 			var feeSum types.Currency
-			size := uint64(len(encoding.Marshal(txn)))
-			for _, fee := range txn.MinerFees {
-				feeSum = feeSum.Add(fee)
+			var sizeSum int
+			for _, txn := range set {
+				sizeSum += len(encoding.Marshal(txn))
```

Comment:
needs to be exact, these numbers are used by the miner for building blocks.

We're already crawling through all the transactions to get the miner fees, might as well encode them at the same time. If there's a meaningful performance difference for some reason we can encode the whole set and then subtract 8l.

Though I guess if we want to start being performance sensitive I should probably be doing the MarshalSia trick here to avoid incurring the T2E cost.